packages/forms/src/__tests__/useAppForm.spec.ts

```diff
@@ -0,0 +1,149 @@
+    Promise.resolve({
+      meta: { status: 200 },
+    }),
+  );
+  const handleSuccess = jest.fn();
+
+  const { result, waitForNextUpdate } = renderHook(() =>
+    useAppForm({
+      initialValues: { foo: 'bar' },
+      onSubmit: handleSubmit,
+      onSubmitSuccess: handleSuccess,
+    }),
+  );
+
+  await act(() => result.current.setFieldValue('foo', 'baz'));
+
+  act(() => result.current.handleSubmit());
+
+  await waitForNextUpdate();
+
+  expect(handleSubmit).toHaveBeenCalledTimes(1);
+  expect(handleSuccess).toHaveBeenCalledTimes(1);
+  expect(handleSuccess).toHaveBeenLastCalledWithMatchingInlineSnapshot(`
+    Array [
+      Object {
+        "meta": Object {
+          "status": 200,
+        },
+      },
+      Object {
+        "foo": "baz",
+      },
+    ]
+  `);
+});
+
+test('handle failure action', async () => {
+  const getFormErrors = jest.fn(() => ({ foo: 'Failed' }));
+  const handleSubmit = jest.fn(() =>
+    Promise.reject({
+      meta: { status: 500 },
+    }),
+  );
+  const handleFailure = jest.fn();
+
+  const { result, waitForNextUpdate } = renderHook(() =>
+    useAppForm({
+      initialValues: { foo: 'bar' },
+      getFormErrors,
+      onSubmit: handleSubmit,
+      onSubmitFailure: handleFailure,
+    }),
+  );
+  expect(result.current.errors).toMatchInlineSnapshot(`Object {}`);
+
+  await act(() => result.current.setFieldValue('foo', 'baz'));
+
+  act(() => result.current.handleSubmit());
+
+  await waitForNextUpdate();
+
+  expect(handleSubmit).toHaveBeenCalledTimes(1);
+  expect(handleFailure).toHaveBeenCalledTimes(1);
+  expect(handleFailure).toHaveBeenLastCalledWithMatchingInlineSnapshot(`
+Array [
+  Object {
+    "meta": Object {
+      "status": 500,
+    },
+  },
+  Object {
+    "foo": "baz",
+  },
+]
+`);
+
+  expect(getFormErrors).toHaveBeenCalledTimes(1);
+  expect(getFormErrors).toHaveBeenLastCalledWithMatchingInlineSnapshot(`
+Array [
+  Object {
+    "meta": Object {
+      "status": 500,
+    },
+  },
+]
+`);
+
+  expect(result.current.errors).toMatchInlineSnapshot(`
+    Object {
+      "foo": "Failed",
+    }
+  `);
+});
+
+test('cancel handler when unmounted', async () => {
+  const handleSubmit = jest.fn(() =>
+    Promise.resolve({
+      meta: { status: 200 },
+    }),
+  );
+  const handleSuccess = jest.fn();
+
+  const { result, unmount } = renderHook(() =>
+    useAppForm({
+      initialValues: { foo: 'bar' },
+      onSubmit: handleSubmit,
+      onSubmitSuccess: handleSuccess,
+    }),
+  );
+
+  act(() => result.current.handleSubmit());
+
+  unmount();
+
+  expect(handleSuccess).toHaveBeenCalledTimes(0);
+});
+
+test('reset form when key changes', async () => {
+  const { result, rerender } = renderHook(
+    (props: Partial<AppFormConfig<{ foo: string }, void>>) =>
+      useAppForm({
+        initialValues: { foo: 'bar' },
+        onSubmit: jest.fn(),
+        ...props,
+      }),
+  );
+
+  await act(() => result.current.setFieldValue('foo', 'baz'));
```

Comment:
```suggestion
  act(() => result.current.setFieldValue('foo', 'baz'));
```

Reply:
Here is an interesting thing, `setFieldValue` returns `Promise` in implementation, however, typing shows that it returns `void`.

Reply:
then make it explicit, 
```typescript
await act(async () => {
})
```